uiService/zkVersion.go

```diff
@@ -233,53 +221,25 @@ func (zks *zkVersionStore) broadcastVersionChange() {
 	}
 }
 
-func (zks *zkVersionStore) startVersionWatch() {
-	if zks.watchState.active {
-		log.Warning("Tried to start version watch while old watch was active")
+func (zks *zkVersionStore) createVersionWatcher() {
+	if zks.versionWatcher != nil {
 		return
 	}
 
-	zks.watchState.Lock()
-	defer zks.watchState.Unlock()
-	zks.watchState.active = true
-	zks.watchState.disconnected = make(chan struct{})
-
-	log.Debug("Version watch started")
-	go zks.pollForVersionChanges()
-}
-
-func (zks *zkVersionStore) stopVersionWatch() {
-	if !zks.watchState.active {
-		log.Warning("Tried to stop version watch with no watch active")
+	watcher, err := zookeeper.CreateValueNodeWatcher(zks.client, zks.versionPath, zks.zkPollingInterval, zks.versionWatcherCallback)
+	if err != nil {
+		// handle error
+		log.WithError(err).Warn("Failed to create ZK node watcher")
```

Comment:
I agree this shouldn't be fatal, but perhaps retried?

Reply:
🤔 good point. it would technically be retried again anytime we regained the connected status, but a more explicit retry like we have in the watchers is a good idea here.